### ISSUE TITLE
CSRF mismatch breaks `neonctl auth`

### ISSUE THREAD

**Issue by salim-b** [neondatabase/neonctl#118]
## Steps to reproduce

```
neonctl auth
```

## Expected result

Successful login (creation of auth token).

## Actual result

When previously logged *out* in browser:

```
/home/salim/.software/managed/npm/lib/node_modules/neonctl/node_modules/openid-client/lib/client.js:443
      throw new OPError(params);
            ^

OPError: request_forbidden (The request is not allowed. The CSRF value from the token does not match the CSRF value from the data store.)
    at Client.callback (/home/salim/.software/managed/npm/lib/node_modules/neonctl/node_modules/openid-client/lib/client.js:443:13)
    at Server.<anonymous> (file:///home/salim/.software/managed/npm/lib/node_modules/neonctl/auth.js:70:52)
    at Server.emit (node:events:514:28)
    at parserOnIncoming (node:_http_server:1143:12)
    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17) {
  error: 'request_forbidden',
  error_description: 'The request is not allowed. The CSRF value from the token does not match the CSRF value from the data store.',
  state: 'naFpFWIoEZP1EJ4o_rkxZcsM6inwqFzojhNWJKPE4yI'
}

Node.js v20.8.0
```

When previously logged *in* in browser (CLI waits indefinitely):

![Bildschirmfoto vom 2023-11-03 01-07-22](https://github.com/neondatabase/neonctl/assets/20040931/c90bf2c3-ee42-48db-9cf0-897cd57426f0)

## Environment

Ubuntu 22.04, Node.js 20.8.0.

Issue occurs with both Chromium and Firefox (vanilla profile).

**Comment by duskpoet** [comment]
Thank you for reporting! I think this might be the problem with oauth setup for email based accounts, we will fix it asap, meanwhile Github auth and Google auth should work properly

**Comment by xenomote** [comment]
Hello @salim-b, I am following up on this bug, just gathering details so that I can diagnose the issue- could you confirm for me, in the browser are you logged in via email+password, or via Gmail/Github/Hasura?

**Comment by salim-b** [comment]
email+password

**Comment by xenomote** [comment]
Great, thank you for confirming. I think we have identified the issue, we will try to keep you updated with our progress in fixing it. Apologies for any inconvenience in the meantime

**Comment by xenomote** [comment]
Hello again @salim-b. I believe the issue around third party authorization in the browser should be fixed now, please retry and let us know if you still have any issues :smiley:

**Comment by salim-b** [comment]
Thanks for the reminder, @xenomote. I can confirm that the issue is fixed in v1.23.3 ☺️
